docs/debug.md

```diff
@@ -0,0 +1,650 @@
+      until Ubuntu Core supports it)
+    * the developer is trying to copy files from `SNAP_APP_PATH` to
+      `SNAP_APP_DATA_PATH` (eg, for write access of a configuration files) and
+      attempts to use a `cp -a`. This results in a `seccomp` failure when the
+      command is run as root because '-a' attempts to copy the ownership
+      (`chown`) of the files in `SNAP_APP_PATH`, but these are owned by an
+      unprivileged system user. To remedy, ship the files `SNAP_APP_PATH` as
+      world-readable, then use `cp -r --preserve=mode` to copy them instead,
+      and optionally adjust the permissions with chmod after copying
+    * the snap ships a `setuid`/`setgid` application. This is currently not
+      supported and security policy will block their use.
+* A snap performs privileged operations that require Linux `capabilities(7)`
+  not granted by the default security policy. This can happen for a number of
+  reasons:
+    * sometimes the app or a dependent library is trying to read a sensitive
+      file in `/proc` or `/sys` that it doesn't actually need to properly
+      function. The app should be made to either not access the file or to
+      fail gracefully in the event of a permission denied error
+      *Note:* the kernel may report a spurious `net_admin denial` that can
+      most likely be ignored (unless your app is modifying routing tables,
+      firewall rules, etc)
+    * some apps require certain kernel functionality to be present and might
+      attempt to load kernel modules. Because loading arbitrary kernel modules
+      would allow the app to escape confinement, this is not allowed by
+      security policy. Use `snappy config ubuntu-core` and add/adjust the
+      `load-kernel-modules` line accordingly when developing your snap. When
+      ready for production with a gadget snap, make sure the modules you need
+      are loaded there. Eg:
+      ```yaml
+      config:
+        ubuntu-core:
+          ...
+          load-kernel-modules: [ module1, module2, ... ]
+      ```
+
+#### Porting applications
+
+When porting an existing app to Ubuntu Core, it is important to understand
+the snappy FHS and how security policy is implemented. In general, to work
+within the security policy template system, you will want to make sure your
+app:
+
+* can be made to work within the application-specific directories
+* does not try to change ownership of files
+* does not change user/group
+* does not try to setup a more lenient `seccomp` filter than that provided by
+  the policy
+
+### Debugging
+When resolving sandbox issues, the first thing you should do is disable
+kernel rate limiting, otherwise the kernel may choose to not log important
+information needed for debugging (even with this, the kernel may still drop
+log messages (rarely)-- if you feel this is the case, reboot and try again):
+
+    $ sudo snappy install snappy-debug
+    $ sudo snappy-debug.security disable-rate-limiting
```

Comment:
You are right about 15.04. Shall we document the behaviour you see and treat rolling as a bug? It looks likely to me. @sergiusens - what do you think?